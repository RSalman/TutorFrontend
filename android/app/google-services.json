{
  "project_info": {
    "project_number": "779865862095",
    "firebase_url": "https://prospr-8a50d.firebaseio.com",
    "project_id": "prospr-8a50d",
    "storage_bucket": "prospr-8a50d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:779865862095:android:1b83d4b60becf7dd",
        "android_client_info": {
          "package_name": "com.tutorfrontend"
        }
      },
      "oauth_client": [
        {
          "client_id": "779865862095-4tk4rhhj654t9tepp8c6i5elqlaj6sgg.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBDjtDNEYE5WbvPy3VumSoq1dEDhKFcOH8"
        }
      ],
      "services": {
        "analytics_service": {
          "status": 1
        },
        "appinvite_service": {
          "status": 1,
          "other_platform_oauth_client": []
        },
        "ads_service": {
          "status": 2
        }
      }
    }
  ],
  "configuration_version": "1"
}