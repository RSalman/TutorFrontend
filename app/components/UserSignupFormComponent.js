import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { Actions } from 'react-native-router-flux';
import t from 'tcomb-form-native';
import _ from 'lodash';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Button } from 'react-native-elements';
import { updateForm, setProgressBar } from '../actions/signup';

class UserSignupFormComponent extends Component {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    this.props.setProgressBar(0.25);
  }

  onPress() {
    var value = this.signup_form.getValue();
    if (value) {/* if validation fails, value will be null */
      Actions.phoneVerification();
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Form
          ref={(form) => { this.signup_form = form; }}
          type={User}
          options={options}
          onChange={(formData) => this.props.updateForm(formData)}
          value={this.props.signup_data}
        />
        <Button raised title="Next" onPress={() => this.onPress()} backgroundColor={buttonColor} />
      </View>
    );
  }
}

var Form = t.form.Form;

var PhoneNumber = t.refinement(t.Number, function(number) {
  return number.toString().length === 10;
});

var ValidEmail = t.refinement(t.String, function(email) {
  var re = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
  return re.test(email);
});

var SecurePassword = t.refinement(t.String, function(password) {
  return password.length >= 6;
});

var User = t.struct({
  first_name: t.String,
  last_name: t.String,
  email: ValidEmail,
  password: SecurePassword,
  phone_number: PhoneNumber
});

/* Define colours */
const buttonColor = '#FF3366';
const lightGrey = '#D8D8D8';
const brightRed = '#ff7575';
const white = '#FFF';

/* Define Form Stylesheet */
const stylesheet = _.cloneDeep(Form.stylesheet);
const font_size = 14;
const container_height = 30;
const transparentColor = 'transparent';

stylesheet.textbox.normal.color = white;
stylesheet.textbox.error.color = white;
stylesheet.textbox.normal.borderWidth = 0;
stylesheet.textbox.error.borderWidth = 0;
stylesheet.textbox.normal.marginBottom = 0;
stylesheet.textbox.error.marginBottom = 0;
stylesheet.textbox.normal.marginBottom = 5;
stylesheet.textbox.error.marginBottom = 5;
stylesheet.textbox.normal.fontSize = font_size;
stylesheet.textbox.error.fontSize = font_size;
stylesheet.textbox.normal.height = container_height;
stylesheet.textbox.error.height = container_height;

stylesheet.textboxView.normal.borderWidth = 0;
stylesheet.textboxView.error.borderWidth = 0;
stylesheet.textboxView.normal.borderRadius = 0;
stylesheet.textboxView.error.borderRadius = 0;
stylesheet.textboxView.normal.borderBottomWidth = 1;
stylesheet.textboxView.error.borderBottomWidth = 1;
stylesheet.textboxView.normal.height = container_height;
stylesheet.textboxView.error.height = container_height;
stylesheet.textboxView.normal.borderColor = white;
stylesheet.textboxView.error.borderColor = brightRed;

stylesheet.controlLabel.normal.color = lightGrey;
stylesheet.controlLabel.error.color = brightRed;
stylesheet.controlLabel.normal.fontSize = font_size;
stylesheet.controlLabel.error.fontSize = font_size;
stylesheet.controlLabel.normal.backgroundColor = transparentColor;
stylesheet.controlLabel.error.backgroundColor = transparentColor;

stylesheet.errorBlock.fontSize = font_size;
stylesheet.errorBlock.color = brightRed;
stylesheet.errorBlock.backgroundColor = transparentColor;

var options = {
  stylesheet: stylesheet,
  fields: {
    email: { error: 'Please enter a valid email' },
    password: { error: 'Must be a minimum of 6 characters', secureTextEntry: true },
    phone_number: { error: 'Please enter a valid phone number' }
  }
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    marginTop: 0,
    padding: 10,
    marginLeft: 15,
    marginRight: 15
  }
});

const mapStateToProps = (state) => {
  return {
    signup_data: state.signup.signup_data,
    form_page: state.signup.form_page,
    progress_status: state.signup.progress_status
  };
};

const mapDispatchToProps = (dispatch) => bindActionCreators({ updateForm, setProgressBar }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(UserSignupFormComponent);
