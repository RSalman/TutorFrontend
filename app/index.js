import React, { Component } from 'react';
import { StyleSheet,TouchableHighlight, Text, View, Navigator } from 'react-native';

import HomeView from './views/HomeView';
import TutorsComponent from './components/TutorsComponent';
import LeilaFeatureView from './views/LeilaFeatureView';
import SalmanFeatureView from './views/SalmanFeatureView';
import MuraadFeatureView from './views/MuraadFeatureView';

export default class TutoringApp extends Component {
  render() {
    return (
  		<Navigator
        	initialRoute={{screen: 'HomeView'}}
        	renderScene={(route, nav) => {return this.renderScene(route, nav)}}  />
    );
  }

 renderScene(route,nav) {
    switch (route.screen) {
      	case "HomeView":
        	return <HomeView navigator={nav} />
      	case "TutorsComponent":
        	return <TutorsComponent navigator={nav} />
      	case "SalmanFeature":
        	return <SalmanFeatureView navigator={nav} /> 
        case "LeilaFeature":
        	return <LeilaFeatureView navigator={nav} /> 
        case "MuraadFeature":
        	return <MuraadFeatureView navigator={nav} /> 
      }
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});
