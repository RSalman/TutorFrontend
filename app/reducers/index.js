import { combineReducers } from 'redux';
import tutors from './tutors';
import profile from './profile';

const rootReducer = combineReducers({
  tutors: tutors,
  profile: profile
});

export default rootReducer;
