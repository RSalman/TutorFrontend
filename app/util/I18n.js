import I18n from 'react-native-i18n';

import en from './locales/en';
import fr from './locales/fr';

I18n.fallbacks = true;

I18n.translations = {
  en,
  fr
};

global.I18n = I18n;
