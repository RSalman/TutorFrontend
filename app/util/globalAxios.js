import axios from 'axios';
import { Alert, NetInfo } from 'react-native';

axios.defaults.baseURL = 'http://cpserver.eastus.cloudapp.azure.com:2000/api/v1';
axios.defaults.headers.post['Content-Type'] = 'application/json';
axios.defaults.headers.get['Content-Type'] = 'application/json';
axios.defaults.headers.post.Accept = 'application/json';
axios.defaults.headers.get.Accept = 'application/json';

var connection = true;
function connectivityChange(isConnected) {
  connection = isConnected;
}

function showNoInternetDialog(config) {
  return new Promise(function(resolve, reject) {
    Alert.alert(
      I18n.t('globalWifiDialog.noInternetConnection'),
      I18n.t('globalWifiDialog.fixInternet'),
      [
        { text: I18n.t('retry'), onPress: () => resolve(axios(config)) },
      ],
      { cancelable: false }
    );
  });
}

NetInfo.isConnected.addEventListener('change', connectivityChange);

axios.interceptors.response.use(undefined, function(error) {
  return connection ? Promise.reject(error) : showNoInternetDialog(error.config);
});

global.axios = axios;
