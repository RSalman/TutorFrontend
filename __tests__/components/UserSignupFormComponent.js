import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import UserSignupFormComponent from '../../app/components/UserSignupFormComponent';
import '../../app/index';
import '../../app/util/globalAxios';
import '../../app/util/I18n';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

function setup(state) {
  const store = mockStore(state);
  const props = {
    setProgressBar: jest.fn(),
    store
  };
  return props;
}

describe('<UserSignupFormComponent />', () => {
  it('renders correctly', () => {
    const props = setup({
      signup: {
       phoneVerified: false,
        phoneVerifying: false,
        signup_data: {
          first_name: 'John',
          last_name: 'Doe',
          email: 'jdoe@uottawa.ca',
          password: '',
          phone_number: ''
        }
      }
    });
    const tree = renderer.create(
      <UserSignupFormComponent {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
