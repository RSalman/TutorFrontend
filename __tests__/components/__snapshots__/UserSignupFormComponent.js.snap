exports[`<UserSignupFormComponent /> renders correctly 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "marginLeft": 15,
      "marginRight": 15,
      "marginTop": 0,
      "padding": 10,
    }
  }>
  <View
    style={Object {}}>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#D8D8D8",
            "fontSize": 14,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }>
        First name
      </Text>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#FFF",
            "borderRadius": 0,
            "borderWidth": 0,
            "height": 30,
          }
        }>
        <TextInput
          accessibilityLabel="First name"
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 0,
              "color": "#FFF",
              "fontSize": 14,
              "height": 30,
              "marginBottom": 5,
              "padding": 7,
            }
          }
          underlineColorAndroid="transparent"
          value="John" />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#D8D8D8",
            "fontSize": 14,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }>
        Last name
      </Text>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#FFF",
            "borderRadius": 0,
            "borderWidth": 0,
            "height": 30,
          }
        }>
        <TextInput
          accessibilityLabel="Last name"
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 0,
              "color": "#FFF",
              "fontSize": 14,
              "height": 30,
              "marginBottom": 5,
              "padding": 7,
            }
          }
          underlineColorAndroid="transparent"
          value="Doe" />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#D8D8D8",
            "fontSize": 14,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }>
        Email
      </Text>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#FFF",
            "borderRadius": 0,
            "borderWidth": 0,
            "height": 30,
          }
        }>
        <TextInput
          accessibilityLabel="Email"
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 0,
              "color": "#FFF",
              "fontSize": 14,
              "height": 30,
              "marginBottom": 5,
              "padding": 7,
            }
          }
          underlineColorAndroid="transparent"
          value="jdoe@uottawa.ca" />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#D8D8D8",
            "fontSize": 14,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }>
        Password
      </Text>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#FFF",
            "borderRadius": 0,
            "borderWidth": 0,
            "height": 30,
          }
        }>
        <TextInput
          accessibilityLabel="Password"
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={true}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 0,
              "color": "#FFF",
              "fontSize": 14,
              "height": 30,
              "marginBottom": 5,
              "padding": 7,
            }
          }
          underlineColorAndroid="transparent"
          value="" />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#D8D8D8",
            "fontSize": 14,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }>
        Phone number
      </Text>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#FFF",
            "borderRadius": 0,
            "borderWidth": 0,
            "height": 30,
          }
        }>
        <TextInput
          accessibilityLabel="Phone number"
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType="numeric"
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 0,
              "color": "#FFF",
              "fontSize": 14,
              "height": 30,
              "marginBottom": 5,
              "padding": 7,
            }
          }
          underlineColorAndroid="transparent"
          value="" />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": 15,
            "marginRight": 15,
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "#FF3366",
          },
          undefined,
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": undefined,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }>
        Next
      </Text>
    </View>
  </View>
</View>
`;
